src/inserter.rs

```diff
@@ -157,12 +159,19 @@ where
     /// # Panics
     /// If called after previous call returned an error.
     #[inline]
-    pub fn write<'a>(&'a mut self, row: &T) -> impl Future<Output = Result<()>> + 'a + Send
+    pub async fn write(&mut self, row: &T) -> Result<()>
     where
         T: Serialize,
     {
         self.uncommitted_entries += 1;
-        self.insert.write(row)
+        let insert = if let Some(insert) = &mut self.insert {
+            insert
+        } else {
+            let mut new_insert = self.client.insert(&self.table)?;
```

Comment:
I believe it should be marked as `unlikely` (i.e. move to `#[cold] #[inline(never)]` function)